### PR TITLE
Favico

### PR DESCRIPTION
**Pull request by captn3m0**

### CODE REVIEW

File: public/update.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+var favicon= new Favico({ // creating a favico instance
```
Comment (captn3m0) [inline public/update.js]: No need to commit `update.js`. Instead commit update.coffee

You can use js2.coffee to translate js to coffee.

File: public/update.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  visibilityChange = "msvisibilitychange";
+} else if (typeof document.webkitHidden !== "undefined") {
+  hidden = "webkitHidden";
+  visibilityChange = "webkitvisibilitychange";
+}
+ 
+var count =0;
+function handleVisibilityChange() {
+  if (document[hidden]) {
+    //handled in polymer.coffee
```
Comment (captn3m0) [inline public/update.js]: When you handle things across different modules, it is considered a good practice to call methods with arguments, with a callback if needed.

File: public/update.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  } else {
+    favicon.reset();
```
Comment (captn3m0) [inline public/update.js]: If there is only an else clause, there is no point in having both if and else. Just keep one.

File: public/update.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+}
+
+//whenever the page is hidden call updateCount
+function updateCount(){
```
Comment (captn3m0) [inline public/update.js]: This should take an argument

File: public/update.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  count++;
+  //document.title="New Messages("+ count+")";  
+  favicon.badge(count);
+}
+
+
+// if browser doesn't support addEventListener or the Page Visibility API
+if (typeof document.addEventListener === "undefined" || 
+  typeof document[hidden] === "undefined") {
+  alert("Page Visibility API is not supported.");
```
Comment (captn3m0) [inline public/update.js]: Don't use alerts. Degrade gracefully.

Ideally, this script should tackle it by switching to title changes.

File: public/index.html [hunk 6]
```diff
@@ -10,12 +10,14 @@
     <meta name="mobile-web-app-capable" content="yes">
     <title>Chat with SDSLabs</title>
 
-    <link rel="icon" sizes="302x302" href="https://i.imgur.com/WNRjxyN.jpg">
+    <link rel="shortcut icon" sizes="302x302" href="jinora.ico" >
```
Comment (captn3m0) [inline public/index.html]: I haven't seen this committed. Also, is the size correct?

File: public/index.html [hunk 7]
```diff
@@ -10,12 +10,14 @@
...
 
+    
     <script src="bower_components/webcomponentsjs/webcomponents.min.js"></script>
     <script src="//ajax.googleapis.com/ajax/libs/jquery/2.1.0/jquery.min.js"></script>
     <script src="//cdnjs.cloudflare.com/ajax/libs/mustache.js/0.8.1/mustache.min.js"></script>
     <script src="/socket.io/socket.io.js"></script>
+    <script src="bower_components/favico.js/favico-0.3.7.min.js"></script>
```
Comment (captn3m0) [inline public/index.html]: You need to update bower.json with the packages that you add, otherwise I can't install them

**Comment by captn3m0** [comment]
Keep pushing commits and fixes to the same branch

File: public/update.js (hunk 1 above)
Comment (anumarathore) [inline public/update.js]: ok

Comment (anumarathore) [inline public/update.js]: ok

File: public/update.js (hunk 3 above)
Comment (anumarathore) [inline public/update.js]: Changed it

File: public/update.js (hunk 5 above)
Comment (anumarathore) [inline public/update.js]: Page visibility API is supported by IE 10+, Chrome and Firefox 10+
and the window.onfocus and window.onblur doesn't work well even in earlier versions. Switching   to title changes must work when the browser doesn't support favico.js  (using bowser).

File: public/index.html (hunk 7 above)
Comment (anumarathore) [inline public/index.html]: Sorry I overlooked it.

**Comment by captn3m0** [comment]
@anumarathore In case you didn't notice, I pushed a commit as well.

**Pull request closed without merging**